.github/workflows/review-quarantine-publish.yml

```diff
@@ -0,0 +1,124 @@
+name: Review Quarantine PR (Publish)
+
+on:
+  # Run only after the generator workflow completes
+  workflow_run:
+    workflows: ["Review Quarantine PR (Generate)"]
+    types: [completed]
+
+permissions:
+  contents: read
+  issues: write          # create/update PR (issue) comments
+  pull-requests: write   # needed to comment on PRs
+
+jobs:
+  publish-comment:
+    name: Publish quarantine comment to PR (only after Generate succeeds)
+    runs-on: ubuntu-24.04
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+
+    permissions:
+      actions: read
+      contents: read
+      issues: write
+      pull-requests: write
+
+    env:
+      # Use workflow_run payload instead of pull_request context
+      REPO: ${{ github.repository }}
+      PR_NUMBER: ${{ github.event.workflow_run.pull_requests[0].number }}
+      HEAD_SHA: ${{ github.event.workflow_run.head_sha }}
+
+    steps:
+      - name: Download generated artifacts from the triggering run
+        uses: actions/download-artifact@v4
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Download artifacts from the exact run that triggered this workflow
+          run-id: ${{ github.event.workflow_run.id }}
+          path: ./quarantine-artifacts
+
+      - name: Read PR number from artifact (fork-safe)
+        id: read_pr
+        shell: bash
+        run: |
+          set -euo pipefail
+          # Find the first pr_number.txt among all artifact subfolders
+          PR_FILE="$(find ./quarantine-artifacts -type f -name 'pr_number.txt' | head -n1 || true)"
+          if [[ -z "${PR_FILE}" ]]; then
+            echo "pr_number=" >> "$GITHUB_OUTPUT"
```

Comment:
why two paths? example how to sanitize: https://github.com/nrfconnect/sdk-nrf/blob/main/.github/workflows/docpublish.yml#L34-L37   (not hugely critical here as artifacts are not zipped but codeql might fail)